Panic on `--disable-cluster-overview`

Immediately when I run

```console
$ octant --disable-cluster-overview
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x169e505]

goroutine 16 [running]:
github.com/vmware-tanzu/octant/internal/kubeconfig.NewKubeConfigContextManager(0x7991820, 0xc0001b5dd0, 0xc0000ca900, 0xd, 0x10, 0xc0000ca900, 0x8, 0x10)
	/__w/octant/octant/internal/kubeconfig/kubeconfig.go:65 +0xd5
github.com/vmware-tanzu/octant/pkg/dash.(*Runner).initAPI(0xc0000a40f0, 0x7991820, 0xc0001b5dd0, 0x79a8d60, 0xc00000d0e0, 0xc0000ce000, 0xd, 0x18, 0x0, 0x0, ...)
	/__w/octant/octant/pkg/dash/dash.go:339 +0x1721
github.com/vmware-tanzu/octant/pkg/dash.(*Runner).apiFromKubeConfig(0xc0000a40f0, 0xc000055360, 0x19, 0xc0000ce000, 0xc, 0x18, 0x7991820, 0xc0001b5dd0, 0x1, 0xc0004923d1, ...)
	/__w/octant/octant/pkg/dash/dash.go:262 +0x173
github.com/vmware-tanzu/octant/pkg/dash.NewRunner(0x7991760, 0xc0001388c0, 0x79a8d60, 0xc00000d0e0, 0xc0000ce000, 0xc, 0x18, 0x3, 0xc00048c240, 0x0)
	/__w/octant/octant/pkg/dash/dash.go:237 +0x7d7
github.com/vmware-tanzu/octant/internal/commands.newOctantCmd.func1.1(0x791b57c, 0x6, 0x79781e0, 0x28, 0x79244d0, 0x14, 0x7989de0, 0xc000488060, 0x7991760, 0xc0001388c0, ...)
	/__w/octant/octant/internal/commands/dash.go:125 +0xa06
created by github.com/vmware-tanzu/octant/internal/commands.newOctantCmd.func1
	/__w/octant/octant/internal/commands/dash.go:76 +0x5ed
```

it work fine if I don't use that cli flag

**Environment:**

- Octant version (use `octant version`): `0.17.0`
- Kubernetes version (use `kubectl version`): `1.18`
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): `Ubuntu 20.04`

--- a/changelogs/unreleased/2058-mklanjsek
+++ b/changelogs/unreleased/2058-mklanjsek
@@ -1,0 +1,1 @@
+Fixed issue with clusterless configuration specified at startup (#2058)

--- a/internal/kubeconfig/kubeconfig.go
+++ b/internal/kubeconfig/kubeconfig.go
@@ -62,8 +62,10 @@ func NewKubeConfigContextManager(ctx context.Context, opts ...KubeConfigOption) 
 	options := kubeConfigOptions{}
 	clusterOptions := []cluster.ClusterOption{}
 	for _, opt := range opts {
-		opt.kubeConfigOption(&options)
-		clusterOptions = append(clusterOptions, opt.clusterOption)
+		if opt.kubeConfigOption != nil {
+			opt.kubeConfigOption(&options)
+			clusterOptions = append(clusterOptions, opt.clusterOption)
+		}
 	}
 	chain := strings.Deduplicate(filepath.SplitList(options.KubeConfigList))
 	rules := &clientcmd.ClientConfigLoadingRules{

--- a/internal/kubeconfig/kubeconfig_test.go
+++ b/internal/kubeconfig/kubeconfig_test.go
@@ -97,3 +97,15 @@ func TestFSLoader_Load(t *testing.T) {
 		{Name: "exp-scratch"},
 	}, kc.Contexts())
 }
+
+func Test_NewKubeConfigNoCluster(t *testing.T) {
+	noClusterOptions := KubeConfigOption{nil, nil}
+
+	_, err := NewKubeConfigContextManager(
+		context.TODO(),
+		WithKubeConfigList(filepath.Join("testdata", "kubeconfig.yaml")),
+		FromClusterOption(cluster.WithRESTConfigOptions(cluster.RESTConfigOptions{})),
+		noClusterOptions,
+	)
+	require.NoError(t, err)
+}

--- a/pkg/dash/dash.go
+++ b/pkg/dash/dash.go
@@ -75,6 +75,7 @@ type RunnerOption struct {
 
 func WithOpenCensus() RunnerOption {
 	return RunnerOption{
+		kubeConfigOption: kubeconfig.Noop(),
 		nonClusterOption: func(o *Options) {
 			o.EnableOpenCensus = true
 		},
@@ -83,6 +84,7 @@ func WithOpenCensus() RunnerOption {
 
 func WithoutClusterOverview() RunnerOption {
 	return RunnerOption{
+		kubeConfigOption: kubeconfig.Noop(),
 		nonClusterOption: func(o *Options) {
 			o.DisableClusterOverview = true
 		},

